### DISCUSSION TITLE
Status API should also support checking of payment reference info

### DISCUSSION THREAD

**Discussion by BLR-0118** [beckn/protocol-specifications#287] [Core Specification]
/status only accepts order_id as per the current spec. This should be enhanced to also support other types of status e.g. for "payment reference info" as per the payment contract sequence defined [here](https://github.com/beckn/protocol-specifications/blob/master/docs/protocol-drafts/BECKN-RFC-002-Payments-On-Beckn-Enabled-Networks.md)

**Comment by ravi-prakash-v** [comment]
`status` fetches status against a `transaction_id`. All other attributes are optional. The BPP is supposed to fetch the latest status of an order using the `transaction_id`.

Can you provide a use case where this is breaking?

**Comment by BLR-0118** [comment]
Check the last 2 steps in the "Payment Contract Sequence" in the []([doc](https://github.com/beckn/protocol-specifications/blob/master/docs/protocol-drafts/BECKN-RFC-002-Payments-On-Beckn-Enabled-Networks.md))
How will this work with /status and /on_status?
